addons/iap/models/iap_account.py

```diff
@@ -19,9 +20,95 @@ class IapAccount(models.Model):
     _rec_name = 'service_name'
     _description = 'IAP Account'
 
-    service_name = fields.Char()
-    account_token = fields.Char(default=lambda s: uuid.uuid4().hex)
+    service_name = fields.Char(string='Account Name')
```

Comment:
Why are we using the `service_name` field as the account name ?
We should have a regular `name` field, remove `_rec_name = 'service_name'` and rename `iap_service_name` to `service_name`, it would be much less confusing.